File: src/FoodSharing.Site/ClientApp/src/domain/announcements/announcementBlank.ts
import { Announcement } from "./announcement";

export class AnnouncementBlank {
    constructor(
        public id: string | null,
        public name: string | null,
        public ownerUserId: string | null,
        public description: string | null,
        public categoryId: string | null,
        public gramsWeight: number | null,
        public cityId: string | null,
        public imagesUrls: string[],
        public uploadPhotos: File[]
    ) { }

    public static Empty(userId: string) {
        return new AnnouncementBlank(null, null, userId, null, null, null, null, [], []);
    }
}

export function mapToAnnouncementBlank(announcement: Announcement) {
    return new AnnouncementBlank(
        announcement.id, announcement.name, announcement.ownerUserId, announcement.description,
        announcement.categoryId, announcement.gramsWeight, announcement.cityId, announcement.imagesUrls, []
    );
}

File: src/FoodSharing.Site/Models/Announcements/AnnouncementBlank.cs
#pragma warning disable CS8618
namespace FoodSharing.Site.Models.Announcements;

public partial class AnnouncementBlank
{
    public Guid? Id { get; set; }
    public String? Name { get; set; }
    public Guid? OwnerUserId { get; set; }
    public String? Description { get; set; }
    public Guid? CategoryId { get; set; }
    public Int32? GramsWeight { get; set; }
    public Guid? CityId { get; set; }
    public String[]? ImagesUrls { get; set; }
    public IFormFile[]? UploadPhotos { get; set; }
}

public partial class AnnouncementBlank
{
    public class Validated
    {
        public Guid Id { get; }
        public String Name { get; }
        public Guid OwnerUserId { get; }
        public String Description { get; }
        public Guid CategoryId { get; }
        public Int32 GramsWeight { get; }
        public Guid? CityId { get;  }
        public List<String> ImagesUrls { get; }
        public IFormFile[] UploadPhotos { get;  }
        public Validated(
            Guid id, String name, Guid ownerUserId, String description,
            Guid categoryId, Int32 gramsWeight, Guid cityId,  String[] imagesUrls, 
            IFormFile[] uploadPhotos)
        {
            Id = id;
            Name = name;
            OwnerUserId = ownerUserId;
            Description = description;
            CategoryId = categoryId;
            GramsWeight = gramsWeight;
            CityId = cityId;
            ImagesUrls = imagesUrls.ToList();
            UploadPhotos = uploadPhotos;
        }
    }
}